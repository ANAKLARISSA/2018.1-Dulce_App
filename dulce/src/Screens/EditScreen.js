import React from 'react';
import {View, TextInput, Text} from 'react-native';
import AGRButton from '../Components/AGRButton';


const styles = {
  container:{
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    backgroundColor: '#FFF',
    paddingHorizontal: 20,
  },
  input:{
    height: 36,
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    marginBottom: 10,
    alignSelf: 'center',
    width: '80%',
    marginTop: 10,
  },
  button: {
    alignItems: 'center',
    backgroundColor: 'pink',
    padding: 10,
    color: 'blue'
  },
};

class EditScreen extends React.Component {
  constructor(props){
    super(props);
    this.state = {editable:false};
  }

  tornarVisivel(){
    this.setState({
      editable: !this.state.editable
    });
  }


  render() {
    return (
      <View style={styles.container}>
        <TextInput style={styles.input}
          placeholder='Nome'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Matricula'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Hospital'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Setor'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Editar senha'
          editable = {this.state.editable}
        />
        <AGRButton
          text ="Editar perfil"
          style={styles.button}
          onPress={() => {this.tornarVisivel()}}
        />

      </View>
    );
  }
}


export default EditScreen;
