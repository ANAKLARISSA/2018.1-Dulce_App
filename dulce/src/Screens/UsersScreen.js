import React from 'react';
import {
  View,
  Text,
  TextInput,
  Button,
  Platform,
  Alert,
  ScrollView,
  FlatList
} from 'react-native';
import AGRButton from '../Components/AGRButton';

const styles = {
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 15,
  },
  text: {
    fontSize: 48
  }
};

class UsersScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      users: [],
      loading: true
    }
  }

  componentDidMount() {
    const url = 'https://jsonplaceholder.typicode.com/users';
    this.setState({ loading: true });
    fetch(url)
    .then((response) => {
      return response.json();
    })
    .then((json) => {
      console.log(json);
      this.setState({ users: json, loading: false })
    })
  }

  render() {
    console.log(this.state);
    return (
      <ScrollView contentContainerStyle={styles.container}>
        {
          this.state.loading ? (
            <Text style={styles.text}>Carregando...</Text>
          ) : (
            <FlatList
              data={this.state.users}
              keyExtractor={(item) => {
                return item.id.toString();
              }}
              renderItem={(data) => {
                return <AGRButton text={data.item.name}  onPress={() => this.props.navigation.navigate('users')} />
              }}
            />
          )
        }
      </ScrollView>
    );
  }
}

export default UsersScreen;
