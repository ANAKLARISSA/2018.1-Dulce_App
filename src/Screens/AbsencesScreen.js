import React from 'react';
import {
  StyleSheet,
  View,
  PixelRatio,
  TouchableOpacity,
  Image,
  ScrollView
} from 'react-native';
import ScreenHeader from '../Components/ScreenHeader';
import DateRangePicker from '../Components/DateRangePicker';
import AGRButton from '../Components/AGRButton';
import ImagePicker from 'react-native-image-picker';
import {Card, CardItem, Text, Body, Form, Textarea, Picker, Right, Left} from 'native-base';

const XDate = require('xdate');

export default class AbsencesScreen extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      documentSource: null,
      absenceOption: '',
      startDay: {
        year: new XDate().getFullYear(),
        month: new XDate().getMonth() + 1,
        day: new XDate().getDate()
      },
      endDay: {
        year: new XDate().getFullYear(),
        month: new XDate().getMonth() + 1,
        day: new XDate().getDate()
      }
    };
  }

  getDocument(response) {
    /* istanbul ignore console */
    console.log('Response = ', response);

    if (response.didCancel) {
      /* istanbul ignore console */
      console.log('User cancelled photo picker');
    }
    else if (response.error) {
      /* istanbul ignore console */
      console.log('ImagePicker Error: ', response.error);
    }
    else if (response.customButton) {
      /* istanbul ignore console */
      console.log('User tapped custom button: ', response.customButton);
    }
    else {
      let source = {uri: response.uri};

      // You can also display the image using data:
      // let source = { uri: 'data:image/jpeg;base64,' + response.data };

      this.setState({
        documentSource: source
      });
    }
  }

  selectPhotoTapped() {
    const options = {
      quality: 1.0,
      maxWidth: 500,
      maxHeight: 500,
      storageOptions: {
        skipBackup: true
      }
    };

    ImagePicker.showImagePicker(options, (response) => this.getDocument(response));
  }

  onSelectDates(startDay, endDay) {
    this.setState({
      startDay: startDay,
      endDay: endDay
    });
  }

  onSelectAbsence(value) {
    this.setState({
      absenceOption: value
    });
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <ScreenHeader
          title='Abono de Faltas'
        />
        <ScrollView>
          <Text style={styles.title}>
                Selecione o período de abono
          </Text>
          <DateRangePicker
            onSuccess={(startDay, endDay) => this.onSelectDates(startDay, endDay)}
            theme={{markColor: '#5f4b8b', markTextColor: 'white'}}
          />
          <Card>
            <CardItem header bordered>
            <View style={styles.view1}>
              <Text style={styles.date}>
                {this.state.startDay.day}/{this.state.startDay.month}/
                {this.state.startDay.year} até {this.state.endDay.day}/{this.state.endDay.month}/{this.state.endDay.year}
              </Text>
            </View>
            </CardItem>

            <CardItem bordered>
              <Body style={{flexDirection: 'row'}}>
                <Left>
                  <Text>
                    Selecione o abono:
                  </Text>
                </Left>
                <Right>
                  <Picker
                    selectedValue={this.state.absenceOption}
                    style={styles.picker}
                    onValueChange={(itemValue) => this.onSelectAbsence(itemValue)}>
                    <Picker.Item label='Atestado' value='medical leave' />
                    <Picker.Item label='Maternidade' value='maternity leave' />
                    <Picker.Item label='Paternidade' value='paternity leave' />
                  </Picker>
                </Right>
              </Body>
            </CardItem>
            <CardItem bordered>
              <Body style={styles.container}>
                  <TouchableOpacity onPress={this.selectPhotoTapped.bind(this)}>
                    <View style={[styles.document, styles.documentContainer, {marginBottom: 20}]}>
                      { this.state.documentSource === null ? <Text>Selecione uma Foto</Text>
                        : <Image style={styles.document} source={this.state.documentSource} />
                      }
                    </View>
                  </TouchableOpacity>
              </Body>
            </CardItem>

            <CardItem bordered>
              <Body>
                <Form style={styles.form}>
                  <Textarea rowSpan={5} bordered placeholder='Justificativa' />
                </Form>
              </Body>
            </CardItem>

            <CardItem footer bordered>
              <AGRButton
                style={styles.button}
                onPress={() => {}}
                text='Selecionar'
              />
            </CardItem>
          </Card>
        </ScrollView>
      </View>
    );
  }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  },
  documentContainer: {
    borderColor: '#9B9B9B',
    borderWidth: 1 / PixelRatio.get(),
    justifyContent: 'center',
    alignItems: 'center'
  },
  document: {
    borderRadius: 5,
    width: 150,
    height: 150
  },
  title: {
    margin: 5,
    alignSelf: 'center',
    fontSize: 15,
    color: '#5f4b8b',
    backgroundColor: 'white'
  },
  date: {
    color: '#5f4b8b',
    fontSize: 20
  },
  dates: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    marginRight: 5,
    marginLeft: 5,
    marginTop: 10,
    borderColor: '#5f4b8b',
    borderWidth: 2,
    borderRadius: 5
  },
  justify: {
    flex: 1.5,
    backgroundColor: '#ede8f7',
    marginLeft: 5,
    marginRight: 5,
    marginTop: 10,
    borderColor: '#5f4b8b',
    borderWidth: 2,
    borderRadius: 5
  },
  button: {
    width: 200,
    marginTop: 15,
    marginRight: 5,
    marginLeft: 10
  },
  icon: {
    fontSize: 40,
    color: '#5f4b8b'
  },
  view1: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center'
  },
  view2: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  view3: {
    flex: 1,
    alignItems: 'center'
  },
  form: {
    width: 375
  },
  picker: {
    height: 50,
    width: 170
  }
});
