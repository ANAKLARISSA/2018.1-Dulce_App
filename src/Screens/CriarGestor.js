import React from 'react';
import {Text, View, TextInput, Image, Alert, ScrollView } from 'react-native';
import ValidationComponent from 'react-native-form-validator';
import AGRButton from '../Components/AGRButton';


const logo = require('../../assets/img/logo.png');

const styles = {
  logo: {
    alignSelf: 'center',
    marginBottom: 20
  },
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    backgroundColor: '#FFF',
    paddingHorizontal: 50,

  },
  input: {
    height: 36,
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    marginBottom: 10,
    width: '100%',
    alignSelf: 'center',
    marginBottom: 20,
  },
  button: {
    marginTop : 30
  },
  error:{
    color: 'red'
  }
};

export default class CriarGestor extends ValidationComponent {

  constructor(props) {
    super(props);
    this.state = {
      nome: '',
      matricula: '',
      hospital: '',
      setor: '',
      senha: '',
      confirmaSenha: '',
      error: false

    }
  }

  _onPressButton(){
    if(this.validate({
        nome: {minlength:3, required: true},
        matricula: {numbers: true, required: true},
        hospital: {required: true},
        setor: {required: true},
        senha: {minlength:4, maxlength:8, required: true},
        confirmaSenha: {minlength:4, maxlength:8, required: true}
      })){
        //Metodo para mandar para a API estará aqui
        Alert.alert(
          'nome: ' + this.state.nome +
          '\nMatricula: ' + this.state.matricula +
          '\nHospital: ' + this.state.hospital +
          '\nSetor: ' + this.state.setor +
          '\nSenha: ' + this.state.senha +
          '\nConfirmação de Senha: ' + this.state.confirmaSenha
         )
      }
  }

  render() {
    return (
      <View style={styles.container}>
        <Image source={logo}  style={styles.logo} />
        <ScrollView>
        <TextInput
          ref="nome"
          style={styles.input}
          placeholder="Nome Completo"
          underlineColorAndroid="transparent"
          onChangeText={(text) => this.setState({
            nome: text})}
          />
          {this.isFieldInError('nome') && this.getErrorsInField('nome').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <TextInput
          ref="matricula"
          style={styles.input}
          placeholder="Matrícula"
          underlineColorAndroid="transparent"
          onChangeText={(text) => this.setState({
            matricula: text})}
        />
        {this.isFieldInError('matricula') && this.getErrorsInField('matricula').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <TextInput
          ref="hospital"
          style={styles.input}
          placeholder="Hospital"
          underlineColorAndroid="transparent"
          onChangeText={(text) => this.setState({
            hospital: text})}
        />
        {this.isFieldInError('hospital') && this.getErrorsInField('hospital').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <TextInput
          style={styles.input}
          placeholder="Setor"
          underlineColorAndroid="transparent"
          onChangeText={(text) => this.setState({
            setor: text})}
        />

        <TextInput
          ref="senha"
          style={styles.input}
          placeholder="Senha"
          underlineColorAndroid="transparent"
          secureTextEntry
          onChangeText={(text) => this.setState({
            senha: text})}
        />
        {this.isFieldInError('senha') && this.getErrorsInField('senha').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <TextInput
          ref="confirmaSenha"
          style={styles.input}
          placeholder="Confirme sua senha"
          underlineColorAndroid="transparent"
          secureTextEntry
          onChangeText={(text) => this.setState({
            confirmaSenha: text})}
        />
        {this.isFieldInError('confirmaSenha') && this.getErrorsInField('confirmaSenha').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <AGRButton
          style={styles.button}
          text= 'Testar'
          onPress={() => this._onPressButton()}
        />
        </ScrollView>
      </View>
    );
  }
}
