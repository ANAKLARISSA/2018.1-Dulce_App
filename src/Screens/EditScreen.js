import React from 'react';
import {View} from 'react-native';
import ScreenHeader from '../Components/ScreenHeader';
import ValidationComponent from 'react-native-form-validator';
//import store from '../Reducers/store';
//import axios from 'axios';
import {Container} from 'native-base';
import t from 'tcomb-form-native';
import SignUpButton from '../Components/SignUpButton';

const Form = t.form.Form;

const styles = {
  container: {
    justifyContent: 'center',
    marginTop: 10,
    padding: 15,
    backgroundColor: '#FFF'
  },
  text: {
    fontSize: 15,
    marginBottom: 10,
    marginLeft: 10,
    color: 'black',
    fontFamily: 'Raleway'
  },
  button: {
    height: 45,
    paddingHorizontal: 50,
    backgroundColor: '#5f4b8b',
    alignSelf: 'center',
    alignItems: 'center',
    width: 145
  }
};


export default class EditScreen extends ValidationComponent {
  constructor(props) {
    super(props);
    this.Service = t.struct({
      nome: t.String,
      email: t.String,
      senha: t.String,
      confirmarSenha: t.String
    });
    this.state = {
      value: default_state
    };
    this.options = {
      fields: default_field_options,
      stylesheet: formStyles
    };
  }

  onChange(value) {
    this.setState({value});
  }
  render() {
    return (
<<<<<<< HEAD
      <Container>
        <ScreenHeader title='Editar conta' />
        <View style={styles.container}>
          <Form
            ref='form'
            options={this.options}
            value={this.state.value}
            type={this.Service}
            onChange={(v) => this.onChange(v)}
          />
        </View>
        <SignUpButton
          text = 'Editar'
          onPress = {() => {}}
          style = {styles.button}
        />
      </Container>
=======
      <View style={styles.container}>
      <ScreenHeader title='Editar Perfil' icon='arrow-back'/>
        <ScrollView>
          <SmallLogo/>
          <AGRInput style={styles.input} value = {this.state.name} editable = {false} nameLabel='Nome'/>
          {this.isFieldInError('name') && this.getErrorsInField('name').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <AGRInput style={styles.input} value = {this.state.registration} editable = {false}
            placeholder='Matricula' nameLabel='Matrícula'/>
          <AGRInput value = {this.state.hospital} style={styles.input} editable = {this.state.editable}
            nameLabel='Hospital' onChangeText={(text) => this.setState({hospital: text})}/>
          {this.isFieldInError('hospital') && this.getErrorsInField('hospital').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <AGRInput style={styles.input} value = {this.state.sector} editable = {this.state.editable}
            nameLabel='setor' onChangeText={(text) => this.setState({sector: text})}/>
          {this.isFieldInError('sector') && this.getErrorsInField('sector').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <AGRInput style={styles.input} editable = {this.state.editable} nameLabel='Editar senha'
            secureTextEntry onChangeText={(text) => this.setState({password: text})}/>
          {this.isFieldInError('password') && this.getErrorsInField('password').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <View style={styles.alinhar}>
            <AGRButton style={styles.button} text='Salvar' onPress={() => {this.save(); this._onPressButton();} }/>
          </View>
        </ScrollView>
      </View>
>>>>>>> 9930bc4... #386 Adicionando botões nas telas
    );
  }
}
const formStyles = {
  ...Form.stylesheet,
  error: {
    color: 'red',
    fontSize: 14,
    marginBottom: 3,
    fontWeight: '600'
  }
};

const default_field_options = {
  Nome: {
    error: 'Campo obrigatório'
  },
  confirmarSenha: {
    secureTextEntry: true,
    error: 'Campo obrigatório'
  },
  Email: {
    error: 'Campo obrigatório'
  },
  senha: {
    secureTextEntry: true,
    error: 'Campo obrigatório'
  }
};

const default_state = {
  nome: '',
  email: '',
  senha: '',
  confirmarSenha: ''
};
