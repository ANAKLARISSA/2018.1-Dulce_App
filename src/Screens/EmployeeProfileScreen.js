import React from 'react';
import {View , Text} from 'react-native';
import AGRButton from '../Components/AGRButton';

const styles = {

container: {
flex: 1,
flexDirection: 'column',
padding: 50,
backgroundColor:'#FFF'
},
text: {
fontSize: 36,
marginBottom: 15,
alignSelf: 'center'
},
informacoes: {
  alignSelf: 'center'
}

}

class EmployeeProfileScreen extends React.Component{

  constructor(props){
    super(props);
    this.state = {
      employee: [],
      loading: true

    }

  }

  componentDidMount(){

    const url = 'https://jsonplaceholder.typicode.com/users';
    this.setState({loading: true});
    fetch(url)
    .then((response) => {return response.json();})
    .then((json) => {this.setState({employee: json,loading: false});})
  }

  render(){
    return(
      <View style={styles.container}>
      {
        this.state.loading ? (<Text style={styles.text}>Carregando...</Text>) : (
          <View style={styles.informacoes}>
          <Text style={styles.text}>{this.state.employee[0].name}</Text>
          <Text style={styles.text}>CRM/Matricula:</Text>
          <Text style={styles.text}>{this.state.employee[0].address.zipcode}</Text>
          <Text style={styles.text}>setor: Pediatria</Text>
          <Text style={styles.text}>Cirurgi√£o</Text>
          <Text style={styles.text}>Hospital do Gama</Text>
          <View style={{height: 50}} />
          <AGRButton
          text='editar'
          onPress={() => this.props.navigation.navigate('edit')} />
          </View>


        )

      }
      </View>
    )

  }



}

export default EmployeeProfileScreen;
