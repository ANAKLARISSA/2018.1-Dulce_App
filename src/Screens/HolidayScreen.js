import React, {Component} from 'react';
import {CalendarList} from 'react-native-calendars';
export default class HolidayScreen extends Component {

  constructor(props) {
    super(props);
    this.state = {
      startingDate: new Date(),
      endingDate: new Date()
    };
  }

  _onPress(day) {
    if (this.state.startingDate.getFullYear() === 1998) {
      this.setState({startingDate: day}, () => {
        console.log('start');
        console.log(this.state.startingDate.dateString.getFullYear());
      }
            );
    } else {
      this.setState({endingDate: day.dateString}, () => {
        console.log('end: ', this.state.endingDate);
      });
    }

  }

  render() {
    return (
            <CalendarList
                onVisibleMonthsChange={(months) => { console.log('now these months are visible', months); }}
                pastScrollRange={0}
                futureScrollRange={12}
                onDayPress={(day) => { this._onPress(day);} }
                scrollEnabled={true}
                showScrollIndicator={true}
                selected={this.state.startingDay}
                markedDates = {
                {
                  '2012-05-20': {textColor: 'green'},
                  '2012-05-22': {startingDay: true, color: 'green'},
                  '2012-05-23': {selected: true, endingDay: true, color: 'green', textColor: 'gray'},
                  '2012-05-04': {disabled: true, startingDay: true, color: 'green', endingDay: true}
                }
                }
                markingType = { 'period'}
            />
    );
  }
}
