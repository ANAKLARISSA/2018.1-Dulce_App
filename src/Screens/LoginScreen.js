import React from 'react';
import { connect } from 'react-redux';
import {View, ScrollView, TextInput, Image, Alert} from 'react-native';
import AGRButton from '../Components/AGRButton';
import AGRInput from '../Components/AGRInput';
import BotaoTransparente from '../Components/BotaoTransparente';
import axios from 'axios';
import { actionLogin } from '../Actions/currentUser';
import { NavigationActions } from 'react-navigation';

const logo = require('../../assets/img/logo.png');

const styles = {
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    backgroundColor: '#FFF'
  },
  input: {
    height: 36,
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    marginBottom: 10,
    width: '100%',
    alignSelf: 'center',
  },
  text: {
    fontSize: 36,
    color: '#039200'

  },
  logo: {
    alignSelf: 'center',
    marginTop: '20%',
    marginBottom: 40
  }
};

export class LoginScreen extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      matricula: '',
      password: ''
    };
  }

  resetNavigation(targetRoute) {
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [
        NavigationActions.navigate({ routeName: targetRoute }),
      ],
    });
    this.props.navigation.dispatch(resetAction);
  }

  _onPressButton() {
    if (this.state.password === '' || this.state.matricula === '') {
      Alert.alert(
        'Alguns campos ainda estão vazios',
        'Para entrar preencha corretamente os campos de matrícula e senha.'
      );
    } else if (this.state.password.length < 6) {
      Alert.alert(
        'Sua senha é muito curta',
        'A senha deve ter no minimo 6 caracteres.'
      );
    } else {
      //this.login();
      const token_falso = {
        matricula: this.state.matricula,
        senha: this.state.password
      };
      this.props.setCurrentUser(token_falso);
      this.resetNavigation('initial')

    }
  }

  login(){
    axios.post('http://localhost:3000/auth/users', {
      matricula: this.state.matricula, password: this.state.password
    })
      .then((response) => {
        this.props.setCurrentUser(response.data.token, response.data.id);
        this.resetNavigation('initial') })
      .catch((error) => {
        if (error.response) {
          Alert.alert(
            'Erro 1'
          );
          console.log(error.response);
        } else if (error.request) {
          Alert.alert(
            'Erro 2'
          );
          console.log(error.request);
        } else {
          Alert.alert(
            'Erro 3'
          );
          console.log(error.messege);
        }
      })
  }
  render() {
    return (
      <View style={styles.container}>
        <ScrollView>
        <Image source={logo} style={styles.logo} />
        <AGRInput
          autoCapitalize='none'
          placeholder='Matrícula'
          onChangeText={(text) => this.setState({matricula: text})}
        />

        <AGRInput
          placeholder='Senha'
          secureTextEntry
          onChangeText={(text) => {
            return this.setState({ password: text });
          }}
        />

        <AGRButton
          onPress={this._onPressButton.bind(this)}
          text="Entrar"
        />
        </ScrollView>
      </View>
    );
  }
}



const mapStateToProps = (state) => {
  return{
    currentUser:state.currentUser
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setCurrentUser: (api_token, api_id) => {
      const currentUser = {
        token: api_token,
        id: api_id
      };
      return dispatch(actionLogin(currentUser));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginScreen);
