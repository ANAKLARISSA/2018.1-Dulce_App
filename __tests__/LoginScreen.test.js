import React from 'react';
import {LoginScreen} from '../src/Screens/LoginScreen';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
