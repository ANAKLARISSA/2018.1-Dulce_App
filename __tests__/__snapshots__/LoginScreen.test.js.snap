// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
      "padding": 15,
    }
  }
>
  <RCTScrollView>
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/logo.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "marginBottom": 40,
            "marginTop": "20%",
          }
        }
      />
      <View
        style={
          Object {
            "paddingHorizontal": 30,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="MatrÃ­cula"
          style={
            Object {
              "alignSelf": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "height": 36,
              "marginBottom": 10,
              "marginTop": 30,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 30,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Senha"
          secureTextEntry={true}
          style={
            Object {
              "alignSelf": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "height": 36,
              "marginBottom": 10,
              "marginTop": 30,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 30,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "grey",
              "height": 50,
              "justifyContent": "center",
              "marginTop": 30,
              "paddingHorizontal": 50,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
              }
            }
          >
            Entrar
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
